asllib/Interpreter.ml

```diff
@@ -1281,10 +1300,20 @@ module Make (B : Backend.S) (C : Config) = struct
         |: SemanticsRule.FBadArity
     (* End *)
     (* Begin EvalFCall *)
-    | Some { body = SB_ASL body; args = arg_decls; _ } ->
+    | Some { body = SB_ASL body; args = arg_decls; recurse_limit; _ } ->
         (let () = if false then Format.eprintf "Evaluating %s.@." name in
          let scope = B.Scope.new_local name in
-         let env1 = IEnv.{ global = genv; local = empty_scoped scope } in
+         let env1 = IEnv.{ global = genv; local = local_empty_scoped scope } in
+         let* () =
+           let* v_limit = eval_loop_limit env1 recurse_limit in
```

Comment:
having `eval_loop_limit` in the context of a call is strange. Perhaps renamed to `eval_limit`?